jetty-javax-transformation/src/main/java/org/eclipse/jetty/transformer/JavaxWebappClassloader.java

```diff
@@ -0,0 +1,92 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.transformer;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.instrument.IllegalClassFormatException;
+import java.security.ProtectionDomain;
+
+import org.eclipse.jetty.webapp.WebAppClassLoader;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.commons.ClassRemapper;
+import org.objectweb.asm.commons.Remapper;
+
+public class JavaxWebappClassloader extends WebAppClassLoader
+{
+
+    private static final JavaxTransfomer INSTANCE = new JavaxTransfomer();
+
+    public JavaxWebappClassloader(Context context) throws IOException
+    {
+        super(context);
+        addTransformer(INSTANCE);
+    }
+
+    static class JavaxTransfomer implements ClassFileTransformer
+    {
+        @Override
+        public byte[] transform(ClassLoader loader, String className, Class<?> classBeingRedefined, ProtectionDomain protectionDomain, byte[] classfileBuffer) throws IllegalClassFormatException
+        {
+            try
+            {
+                ClassReader cr = new ClassReader(new ByteArrayInputStream(classfileBuffer));
+                ClassWriter cw = new ClassWriter(0);
+
+                ClassRemapper classRemapper = new ClassRemapper(cw, new JavaxRemapper());
+
+                cr.accept(classRemapper, ClassReader.EXPAND_FRAMES);
+                return cw.toByteArray();
+            }
+            catch (IOException e)
+            {
+                IllegalClassFormatException icfe = new IllegalClassFormatException(e.getMessage());
+                icfe.initCause(e);
+                throw icfe;
+            }
+        }
+    }
+
+    static class JavaxRemapper extends Remapper
+    {
+
+        private static final String JAVAX_SERVLET = "javax/servlet/";
+
+        private static final String JAKARTA_SERVLET = "jakarta/servlet/";
```

Comment:
We'll need a more robust mapping eventually.

Here's all of the `javax.*` to `jakarta.*` imports in Jetty itself ... https://gist.github.com/joakime/88909e94d1be776a8bdecc0698fd55eb

Reply:
Do we have to worry about version mapping?
Like, Javax Servlet 2.5 to Jakarta Servlet 5.0? (it will be missing many methods and whatnot)

It might make more sense to only go backwards (from most recent version of jakarta back to javax namespace)

Reply:
Isn't that what this PR is attempting?   It is not giving a 5.0 request renamed to a 2.5 request to a 2.5 app, rather it is renaming the 2.5 app to be a 5.0 app.